src/main/proto/firrtl.proto

```diff
@@ -0,0 +1,451 @@
+syntax = "proto3";
+
+package hardware.chips.chisel3;
+
+option java_package = "hardware.chips.chisel3.firrtl";
+
+message Firrtl {
+  repeated Circuit circuit = 1;
+
+  message SourceInfo {
+    message None {
+      enum Reason {
+        NONE_SOURCE_INFO_REASON_UNKNOWN = 0;
+        NONE_SOURCE_INFO_REASON_UNLOCATABLE = 1;
+        NONE_SOURCE_INFO_REASON_SUPPRESSED = 2;
+        NONE_SOURCE_INFO_REASON_DEPRECATED = 3;
+      }
+
+      // Required.
+      Reason reason = 1;
+    }
+
+    message Position {
+      // Required.
+      string filename = 1;
+      // Required.
+      uint32 line = 2;
+      // Required.
+      uint32 column = 3;
+    }
+
+    // Required.
+    oneof source_info {
+      None none = 1;
+      Position position = 2;
+    }
+  }
+
+  message Top {
+    // Required.
+    string name = 1;
+  }
+
+  message Circuit {
+    repeated Module module = 1;
+    repeated Top top = 2;
+  }
+
+  message Module {
+    message ExternalModule {
+      // Required.
+      string id = 1;
+      repeated Port port = 2;
+    }
+
+    message UserModule {
+      // Required.
+      string id = 1;
+      repeated Port port = 2;
+      repeated Statement statement = 3;
+    }
+
+    // Required.
+    oneof module {
+      ExternalModule external_module = 1;
+      UserModule user_module = 2;
+    }
+  }
+
+  message Statement {
+    message Wire {
+      // Required.
+      string id = 1;
+      // Required.
+      Type type = 2;
+    }
+
+    message Register {
+      // Required.
+      string id = 1;
+      // Required.
+      Type type = 2;
+      // Required.
+      Expression clock = 3;
+      Expression reset = 4;
+      Expression init = 5;
+    }
+
+    message Memory {
+      // Required.
+      string id = 1;
+      // Required.
+      Type type = 2;
+      // Required.
+      uint32 depth = 3;
+      // Required.
+      uint32 write_latency = 4;
+      // Required.
+      uint32 read_latency = 5;
+      repeated string reader_id = 6;
+      repeated string writer_id = 7;
+    }
+
+    message CMemory {
+      // Required.
+      string id = 1;
+      // Required.
+      Type.VectorType type = 2;
+    }
+
+    message Instance {
+      // Required.
+      string id = 1;
+      // Required.
+      string module_id = 2;
+    }
+
+    message Node {
+      // Required.
+      string id = 1;
+      // Required.
+      Expression expression = 2;
+    }
+
+    message When {
+      // Required.
+      Expression predicate = 1;
+      repeated Statement consequent = 2;
+      repeated Statement otherwise = 3;
+    }
+
+    message Stop {
+      // Required.
+      int32 return_value = 1;
+      // Required.
+      Expression clk = 2;
+      // Required.
+      Expression en = 3;
+    }
+
+    message Printf {
+      // Required.
+      bytes value = 1;
+      repeated Expression arg = 2;
+      // Required.
+      Expression clk = 3;
+      // Required.
+      Expression en = 4;
+    }
+
+    message InlineVerilog {
```

Comment:
InlineVerilog, Assert, and Cover haven't been added yet, so perhaps just comment them out with a TODO? Or add an issue to discuss these new constructs, and reference the issue in the comment?

Reply:
Deleting for now.